4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-general-gui-handle-complex-gui-easily-with-recursive-overrides.md

---
type: entry
section: general
subsection: GUI
rank: 2
importance: 3
anchor: handle-complex-gui-easily-with-recursive-overrides

> title: Handle complex GUI easily with recursive overrides

Reviewer: ```suggestion
title: Handle complex GUIs easily with recursive overrides
```

Reviewer 2: I would keep the singular, or use "Handle complex UI easily" (we use UI for this in the documentation etc.)

Reviewer: If singular, it should be "Handle **_a_** complex GUI easily...".

blockquote: Gooey GUI no more
text: |

>   It is now possible to change mouse and focus behavior of a `Control` node recursively. This greatly helps creating complex GUI without breaking a sweat.

Reviewer: ```suggestion
  It is now possible to change mouse and focus behavior of a `Control` node recursively. This greatly helps creating complex GUIs without breaking a sweat.
```

>   Here's an example: let's say you create an inventory screen for your game. On the left, there's a grid displaying what's the hero is carrying. On the right, it shows a detailed view about the selected item on the left; a rotatable display of the item in 3D to examine every detail, a section containing a scrollable description, a box containing stats and modifiers (with hyperlinks for technical terms), and a list of buttons representing actions that are possible to do with it.

Reviewer 2: ```suggestion
  Here’s an example: let’s say you create an inventory screen for your game. On the left, there’s a grid displaying what the hero is carrying. On the right, it shows a detailed view about the selected item on the left; a rotatable display of the item in 3D to examine every detail, a section containing a scrollable description, a box containing stats and modifiers (with hyperlinks for technical terms), and a list of buttons representing actions that are possible to do with it.
```

>   Now, the problem is that the right view depends on the selection of an item on the left. The user shouldn't be able to interact with the detail view until it happens.

Reviewer 2: ```suggestion
  Now, the problem is that the right view depends on the selection of an item on the left. The user shouldn’t be able to interact with the detail view until it happens.
```

>   Now, the problem is that the right view depends on the selection of an item on the left. The user shouldn’t be able to interact with the detail view until it happens.

Reviewer: ```suggestion
  Now, the problem is that the right view depends on an item being selected on the left. The user shouldn’t be able to interact with the detail view until that happens.
```

>   By changing [`Control.focus_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-focus-behavior-recursive) and [`Control.mouse_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-mouse-behavior-recursive) of the detailed view container to their disabled value until an item selection, it now disables focus and mouse events for every children. You don't have to resort anymore to complex messages to manage behavior of `Control` groups.

Reviewer 2: ```suggestion
  By changing [`Control.focus_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-focus-behavior-recursive) and [`Control.mouse_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-mouse-behavior-recursive) of the detailed view container to their disabled value until an item is selected, it now disables focus and mouse events for every child. You don’t have to resort to complex messages anymore to manage the behavior of `Control` groups.
```

>   By changing [`Control.focus_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-focus-behavior-recursive) and [`Control.mouse_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-mouse-behavior-recursive) of the detailed view container to their disabled value until an item is selected, it now disables focus and mouse events for every child. You don’t have to resort to complex messages anymore to manage the behavior of `Control` groups.

Reviewer: ```suggestion
  By changing [`Control.focus_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-focus-behavior-recursive) and [`Control.mouse_behavior_recursive`](https://docs.godotengine.org/en/latest/classes/class_control.html#class-control-property-mouse-behavior-recursive) of the detailed view container to their disabled value until an item is selected, focus and mouse events will be disabled for every child. You no longer have to resort to complex messages to manage the behavior of `Control` groups.
```
